eng/pipelines/ci.yaml

```diff
@@ -32,6 +32,15 @@ jobs:
       vmImage: "MMSUbuntu20.04"
 
     steps:
+      - checkout: self
+        submodules: true
+
+      - script: |
+          git checkout .
+          git apply ../core.diff --ignore-whitespace
+        displayName: 'Patch core'
+        workingDirectory: ./core
```

Comment:
I've put this patch to `Generate-TypeSpec.ps1` (actually this does the Build too).

But I didn't add this to `Generate.ps1` as it does not contain the Build step (it is in follow step in yaml)
```
      - task: Maven@3
        displayName: 'Build and Package'
        inputs:
          mavenPomFile: pom.xml
          goals: 'clean install'
          options: '$(DefaultOptions) -P local -T 1C'
          mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
          javaHomeOption: 'JDKVersion'
          jdkVersionOption: $(JavaVersion)
          jdkArchitectureOption: 'x64'
          publishJUnitResults: false
```

Let me know what's the preference. (usually this patch only need to be done once after submodule init).